src/main/java/net/solmis/cityparking/controllers/GetParkingReceiptController.java

```diff
@@ -0,0 +1,102 @@
+package net.solmis.cityparking.controllers;
+
+import net.solmis.cityparking.Payment;
+import net.solmis.cityparking.RateCalculator;
+import net.solmis.cityparking.Ticket;
+import net.solmis.cityparking.exceptions.CurrencyNotSupportedException;
+import net.solmis.cityparking.requests.ErrorResponse;
+import net.solmis.cityparking.requests.GetParkingReceiptRequest;
+import net.solmis.cityparking.requests.GetParkingReceiptResponse;
+import net.solmis.cityparking.requests.Response;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.time.ZonedDateTime;
+import java.util.Currency;
+
+@RestController
+public class GetParkingReceiptController {
+
+    @Value("${config.defaultCurrency}")
+    private String defaultCurrencyCode;
+
+    @PostMapping("/getParkingReceipt")
+    public ResponseEntity<Response> getParkingReceipt(@Valid @RequestBody GetParkingReceiptRequest request) {
+
+        Ticket correspondingTicket = Ticket.get(request.ticketId);
+        if (correspondingTicket == null)
+            return new ResponseEntity<>(new ErrorResponse("No corresponding ticket for given ID."),
+                    HttpStatus.BAD_REQUEST);
+        if (correspondingTicket.isActive())
+            return new ResponseEntity<>(new ErrorResponse("First end parking before trying to get a receipt."),
+                    HttpStatus.BAD_REQUEST);
+        if (!correctDriverType(request.driverType))
+            return new ResponseEntity<>(new ErrorResponse("Incorrect driver type."), HttpStatus.BAD_REQUEST);
+
+        Currency defaultCurrency = Currency.getInstance(defaultCurrencyCode);
+        try {
+            GetParkingReceiptResponse response = createResponse(correspondingTicket, request.driverType, defaultCurrency);
+            createFakePayment(response);
+            return new ResponseEntity<>(response, HttpStatus.OK);
+        } catch (CurrencyNotSupportedException e) {
+            return new ResponseEntity<>(new ErrorResponse("Currency not supported."), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    private GetParkingReceiptResponse createResponse(Ticket correspondingTicket, String driverType,
+                                                     Currency currency) throws CurrencyNotSupportedException {
+        if (driverType.equals(GetParkingReceiptRequest.DRIVER_REGULAR))
+            return createForRegular(correspondingTicket, currency);
+        else if (driverType.equals(GetParkingReceiptRequest.DRIVER_DISABLED))
+            return createForDisabled(correspondingTicket, currency);
+        return null;
```

Comment:
i co w takim przypadku się stanie?

Reply:
To ewidentne przeoczenie z mojej strony. Napisałem coś najpierw i później zapomniałem zmienić. Można tam rzucić nowo stworzony wyjątek np. InvalidDriverTypeException, złapać w getParkingReceipt i zwrócić stosowny bad request.